### PR TITLE
allow removing of code snippet

### LINKED ISSUE #309: Allow a code snippet to be removed
Currently, there is no UI element to remove a code snippet when creating or editing a test. A button that removes the DOM elements should do the trick.

### PR DESCRIPTION
**Pull request by maxbeatty**
closes #309 

This is the _easier_ approach to allowing a code snippet to be removed although it would be nice to have a button or something. All of the server logic is already in place to deal with a deletion this way.

### CODE REVIEW

File: server/lib/schema.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ exports.testPage = Joi.object().keys({
   setup: mediumText,
   teardown: mediumText,
   test: Joi.array().required().min(2).items(Joi.object().required().keys({
-    title: Joi.string().required().trim().allow('').empty('').max(255).when('code', {
```
Comment (maxbeatty) [inline server/lib/schema.js]: `required` and `allow('')` apparently don't mix

Comment (mathiasbynens) [inline server/lib/schema.js]: Hah! Good catch!

**Comment by mathiasbynens** [comment]
LGTM with build failure addressed

**Comment by maxbeatty** [comment]
Turns out this was more complicated because of the circular dependencies in Joi (https://github.com/hapijs/joi/issues/588 https://github.com/hapijs/joi/issues/1236). Added more tests to verify behavior.

**Pull request merged**